File: app/src/main/java/code/name/monkey/retromusic/name/BTAudioStateReceiver.java
public class BTAudioStateReceiver extends BroadcastReceiver {
    private AudioManager audioManager;
    private int originalVolume = -1;

    public void setAudioManager(AudioManager manager) {
        this.audioManager = manager;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction() == null) return;

        if (BluetoothHeadset.ACTION_AUDIO_STATE_CHANGED.equals(intent.getAction())) {
            int state = intent.getIntExtra(BluetoothHeadset.EXTRA_STATE, BluetoothHeadset.STATE_DISCONNECTED);

            if (audioManager == null) return;

            if (state == BluetoothHeadset.STATE_CONNECTED || state == BluetoothHeadset.STATE_CONNECTING) {
                // 蓝牙正在播放音频，降低音量
                lowerVolume();
            } else if (state == BluetoothHeadset.STATE_DISCONNECTED) {
                // 蓝牙断开，恢复音量
                restoreVolume();
            }
        }
    }

    private void lowerVolume() {
        if (originalVolume == -1) {
            originalVolume = audioManager.getStreamVolume(AudioManager.STREAM_MUSIC);
        }
        int newVolume = Math.max(originalVolume - 2, 0); // 音量降低2级
        audioManager.setStreamVolume(AudioManager.STREAM_MUSIC, newVolume, 0);
    }

    private void restoreVolume() {
        if (originalVolume != -1) {
            audioManager.setStreamVolume(AudioManager.STREAM_MUSIC, originalVolume, 0);
            originalVolume = -1; // 重置
        }
    }
}

File: app/src/main/java/code/name/monkey/retromusic/BTAudioStateReceiver.kt
// 文件路径: app/src/main/java/code/name/monkey/retromusic/BTAudioStateReceiver.kt

package code.name.monkey.retromusic

import android.bluetooth.BluetoothHeadset
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.media.AudioManager

class BTAudioStateReceiver : BroadcastReceiver() {

    private var audioManager: AudioManager? = null
    private var originalVolume = -1

    fun setAudioManager(manager: AudioManager) {
        audioManager = manager
    }

    override fun onReceive(context: Context?, intent: Intent?) {
        if (intent?.action != BluetoothHeadset.ACTION_AUDIO_STATE_CHANGED) return

        val state = intent.getIntExtra(BluetoothHeadset.EXTRA_STATE, BluetoothHeadset.STATE_DISCONNECTED)
        val am = audioManager ?: return

        when (state) {
            BluetoothHeadset.STATE_CONNECTED,
            BluetoothHeadset.STATE_CONNECTING -> lowerVolume(am)
            BluetoothHeadset.STATE_DISCONNECTED -> restoreVolume(am)
        }
    }

    private fun lowerVolume(am: AudioManager) {
        if (originalVolume == -1) {
            originalVolume = am.getStreamVolume(AudioManager.STREAM_MUSIC)
        }
        val newVolume = Math.max(originalVolume - 2, 0)
        am.setStreamVolume(AudioManager.STREAM_MUSIC, newVolume, 0)
    }

    private fun restoreVolume(am: AudioManager) {
        if (originalVolume != -1) {
            am.setStreamVolume(AudioManager.STREAM_MUSIC, originalVolume, 0)
            originalVolume = -1
        }
    }
}
